### PR TITLE
CUMULUS-1485: Update cmr-client to return error message from CMR for validation failures

### PR DESCRIPTION
**Pull request by jennyhliu**
**Summary:** Summary of changes

Addresses [CUMULUS-1485: Post-to-cmr task to return error message from CMR](https://bugs.earthdata.nasa.gov/browse/CUMULUS-1485)

## Changes

* Update `@cumulus/cmr-client` to return error message from CMR for validation failures.
* Add unit tests

### CODE REVIEW

File: packages/cmr-client/tests/test-validate.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+  process.env.CMR_ENVIRONMENT = 'UAT';
+
+  try {
+    await validateUMMG(ummMetadata, granuleId, provider);
+    t.fail('Expected a validation error to be thrown');
+  } catch (e) {
+    t.true(e instanceof ValidationError);
+    t.true(e.message.includes(cmrError));
+    t.pass();
```
Comment (yjpa7145) [inline packages/cmr-client/tests/test-validate.js]: This `t.pass()` is not necessary, there are already two assertions that would have been run.

File: packages/cmr-client/tests/test-validate.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+  process.env.CMR_ENVIRONMENT = 'UAT';
+
+  try {
+    await validate('granule', xmlMetadata, granuleId, provider);
+    t.fail('Expected a validation error to be thrown');
+  } catch (e) {
+    t.true(e instanceof ValidationError);
+    t.true(e.message.includes(cmrError));
+    t.pass();
```
Comment (yjpa7145) [inline packages/cmr-client/tests/test-validate.js]: Same comment about this `t.pass()`

**Comment by yjpa7145** [comment]
> After talking to @kkelly51 about it, I think it makes sense to pull `validateUMMG` out into its own JS module. I'll issue a PR that does that, and pulls some tests into cmr-client that were originally in cmrjs.

Here is the PR that moves `validateUMMG`, and also adds some more tests: https://github.com/nasa/cumulus/pull/1225

**Comment by jennyhliu** [comment]
> > After talking to @kkelly51 about it, I think it makes sense to pull `validateUMMG` out into its own JS module. I'll issue a PR that does that, and pulls some tests into cmr-client that were originally in cmrjs.
> 
> Here is the PR that moves `validateUMMG`, and also adds some more tests: #1225

Thanks!

**Pull request merged**